services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -911,10 +916,32 @@ class HakemusService(
         if (updatedApplicationEntity.hanke.generated) {
             updatedApplicationEntity.hanke.nimi = request.name
         }
+        updateTormaystarkastelut(updatedApplicationEntity.hakemusEntityData)
         sendApplicationNotifications(updatedApplicationEntity, originalContactUserIds, userId)
         return hakemusRepository.save(updatedApplicationEntity)
     }
 
+    /** Calculate the traffic nuisance indexes for each work area of an application. */
+    private fun updateTormaystarkastelut(hakemusEntityData: HakemusEntityData) {
+        if (hakemusEntityData is KaivuilmoitusEntityData &&
+            hakemusEntityData.startTime != null &&
+            hakemusEntityData.endTime != null) {
+            hakemusEntityData.areas?.forEach { area ->
+                area.tyoalueet.forEach { tyoalue ->
+                    tyoalue.tormaystarkasteluTulos =
+                        tormaystarkasteluLaskentaService.calculateTormaystarkastelu(
```

Comment:
This method is nested deep. It should be split.

Reply:
I splitted this.